tests/functions/report.nf.test

```diff
@@ -0,0 +1,202 @@
+        }
+    }
+
+    test("Test tab missing 2 column values, header supplied"){
+
+        when {
+            function {
+                """
+                input[0] = file("$baseDir/tests/data/tables/mock_missing_value_2.tab")
+                input[1] = true
+                input[2] = '\t'
+                input[3] = null
+                """
+            }
+            params {
+                outdir = "results"
+            }
+        }
+        then{
+            assert function.result == ['0':['header1':'NoData', 'header2':'NoData', 'header3':'stuff3']]
+            assert function.success
+        }
+    }
+
+
+    test("Test tab, header supplied"){
+
+        when {
+            function {
+                """
+                input[0] = file("$baseDir/tests/data/tables/no_missing.tab")
+                input[1] = true
+                input[2] = '\t'
+                input[3] = null
+                """
+            }
+            params {
+                outdir = "results"
+            }
+        }
+        then{
+            assert function.result == ['0':['header1':'stuff1', 'header2':'stuff2', 'header3':'stuff3']]
+            assert function.success
+        }
+    }
+
+    test("Test csv missing column value, header supplied"){
+
+        when {
+            function {
+                """
+                input[0] = file("$baseDir/tests/data/tables/mock_missing_value.csv")
+                input[1] = true
+                input[2] = ','
+                input[3] = null
+                """
+            }
+            params {
+                outdir = "results"
+            }
+        }
+        then{
+            assert function.result == ['0':['header1':'NoData', 'header2':'stuff2', 'header3':'stuff3']]
+            assert function.success
+        }
+    }
+
+
+    test("Test csv, header supplied"){
+
+        when {
+            function {
+                """
+                input[0] = file("$baseDir/tests/data/tables/no_missing.csv")
+                input[1] = true
+                input[2] = ','
+                input[3] = null
+                """
+            }
+            params {
+                outdir = "results"
+            }
+        }
+        then{
+            assert function.result == ['0':['header1':'stuff1', 'header2':'stuff2', 'header3':'stuff3']]
+            assert function.success
+        }
+    }
+
+    test("Test csv, header missing one value"){
+
+        when {
+            function {
+                """
+                input[0] = file("$baseDir/tests/data/tables/header_missing_val.csv")
+                input[1] = true
+                input[2] = ','
+                input[3] = null
+                """
+            }
+            params {
+                outdir = "results"
+            }
+        }
+        then{
+            assert function.success
+            assert function.result == ['0':['__default_index':'stuff1', 'header2':'stuff2', 'header3':'stuff3']]
```

Comment:
Shouldn't this be `__default_index__`?

Reply:
fixed in: https://github.com/phac-nml/mikrokondo/pull/83/commits/049c43c917af70478960c9ec3f1a83b49b931cf9